packages/core-components/src/components/ProgressBars/Gauge.tsx

```diff
@@ -104,19 +119,41 @@ export const getProgressColor: GaugePropsGetColor = ({
  */
 
 export function Gauge(props: GaugeProps) {
+  // const hoverRef = useRef<HTMLDivElement>(null);
+  const [hoverRef, setHoverRef] = useState<HTMLDivElement | null>(null);
   const { getColor = getProgressColor } = props;
   const classes = useStyles(props);
   const { palette } = useTheme<BackstageTheme>();
-  const { value, fractional, inverse, unit, max } = {
+  const { value, fractional, inverse, unit, max, description } = {
     ...defaultGaugeProps,
     ...props,
   };
 
   const asPercentage = fractional ? Math.round(value * max) : value;
   const asActual = max !== 100 ? Math.round(value) : asPercentage;
 
+  const [isHovering, setIsHovering] = useState(false);
+  const handleMouseOver = () => setIsHovering(true);
+  const handleMouseOut = () => setIsHovering(false);
+
+  useEffect(() => {
+    const node = hoverRef;
+    if (node && description) {
+      node.addEventListener('mouseenter', handleMouseOver);
+      node.addEventListener('mouseleave', handleMouseOut);
+
+      return () => {
+        node.removeEventListener('mouseenter', handleMouseOver);
+        node.removeEventListener('mouseleave', handleMouseOut);
+      };
+    }
+    return () => {
+      setIsHovering(false);
+    };
+  });
```

Comment:
Should add dependencies to this hook so we don't re-run it too often right?

Reply:
@Rugvip  Thank you!! Just making sure I understand. Update the hook to this by adding `handleMouseOver, handleMouseOut`. Full change to this:
`useEffect(() => {
    const node = hoverRef;
    if (node && description) {
      node.addEventListener('mouseenter', handleMouseOver);
      node.addEventListener('mouseleave', handleMouseOut);

      return () => {
        node.removeEventListener('mouseenter', handleMouseOver);
        node.removeEventListener('mouseleave', handleMouseOut);
      };
    }
    return () => {
      setIsHovering(false);
    };
  },[handleMouseOver, handleMouseOut]);`

Reply:
Hmm, so I think you might actually want to move the handler definitions inside the `useEffect` callback. The dependencies I was referring to are `hoverRef` and `description`